virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -224,24 +239,68 @@ def setup(self) -> None:
             )
         )
 
+        # Turn elevation data into np array
+        elevation = np.array(self.data["elevation"])
+
+        # map neighbours and identify downstream cell_ids
+        mapping = [
+            self.data.grid.neighbours[cell_id]
+            for cell_id in range(len(self.data.grid.cell_id))
+        ]
+        downstream_ids = []
+
+        for cell_id, neighbors_id in enumerate(mapping):
+            # Find lowest neighbour
+            downstream_id_loc = np.argmax(elevation[cell_id] - elevation[neighbors_id])
+            downstream_ids.append(neighbors_id[downstream_id_loc])
+
+        # Invert mapping of cell_id: downstream neighbour to cell_id: upstream_neighbors
+        upstream_ids: list = [[] for i in range(len(self.data.grid.cell_id))]
+        for down_s, up_s in enumerate(downstream_ids):
+            upstream_ids[up_s].append(down_s)
+
+        self.upstream_ids = upstream_ids
+        """Upstream ID's for the calculation of accumulated surface runoff."""
+
+        # Get the runoff created by SPLASH or initial data set as the baseline:
+        baseline_runoff = np.array(self.data["surface_runoff"])
+
+        # set initial accumulated runoff to zero
+        accumulated_runoff = np.full_like(self.data["elevation"], 0)
+
+        # calculate accumulated runoff for each cell (sum of the upstream neighbours)
+        # TODO this requires a proper spin up!
+        # TODO check for negative numbers
+
+        for cell_id, upstream_id in enumerate(self.upstream_ids):
+            accumulated_runoff[cell_id] += np.sum(baseline_runoff[upstream_id])
```

Comment:
Is there a reason why these todos can't be done in this PR too? In particular, it seems like it should be easy to check for negative numbers. If you're not sure exactly what you want to do when you detect one, you could just raise an error for now and figure out how to handle it more gracefully in future.

Again, I think this block should be separated out from the rest of this function, either into a function with the rest of the code above or into its own function.

Reply:
This is definitely jumping the gun, but we could have a set of [0,1] coefficients here to allow partial draining into different cells 😄

Reply:
@davidorme partial drainage is something for the below-ground flow I guess, which is not yet implemented. We can discuss that at a later stage ;-)